### PR TITLE
Fix uvloop to run on 3.8

### PR DESCRIPTION
**Pull request by 1st1**
Partially addresses: #277 and #266.

### CODE REVIEW

File: uvloop/_testbase.py [hunk 1]
```diff
@@ -415,7 +415,9 @@ def __init__(self, test, sock, prog, timeout):
     def run(self):
         try:
             self._prog(TestSocketWrapper(self._sock))
-        except Exception as ex:
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except BaseException as ex:
```
Comment (elprans) [inline uvloop/_testbase.py]: Maybe do `except Exception, asyncio.CancelledError` instead.  Catching `BaseException` always makes me uneasy.

Comment (1st1) [reply uvloop/_testbase.py]: Why? We handle `KeyboardInterrupt` and `SystemExit`. Besides, that's how we do it in asyncio now, so there's no option, really :(

File: .travis.yml [hunk 2]
```diff
@@ -66,6 +71,13 @@ matrix:
           python: "3.7"
           env: BUILD=tests
 
+        - os: linux
+          dist: xenial
+          sudo: required
```
Comment (Jamim) [inline .travis.yml]: Hello @1st1,
`sudo` option [is deprecated](https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration).
```suggestion
```

Comment (Jamim) [reply .travis.yml]: @1st1 I see you removed `sudo` options.
Just want to note that one of them [is still here](https://github.com/MagicStack/uvloop/blob/0c2cb54dbbf7a7661b912e76e2ae7d35e68e1922/.travis.yml#L67).

Comment (1st1) [reply .travis.yml]: Thanks!

**Pull request merged**